### PR TITLE
Fix bulk api query results iteration

### PR DESCRIPTION
**Pull request by itymch**
Added a fix, which allows to iterate over bulk API query result IDs

### CODE REVIEW

File: simple_salesforce/tests/test_bulk.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    # Python 2.6
+    import unittest2 as unittest
+except ImportError:
+    import unittest
+
+import responses
+
+try:
+    # Python 2.6/2.7
+    import httplib as http
```
Comment (jankatins) [inline simple_salesforce/tests/test_bulk.py]: Unneded, since support for 2.x was dropped in the most recent release

**Comment by ptnebel** [comment]
what is the timeline for merging this?  Desperately needed.

**Comment by jon-wobken** [comment]
@itymch - Can you-rebase this? I will look to review then merge once done. I like the generator approach to this.

**Comment by wyattshapiro** [comment]
looks like there are several PRs open related to this issue (https://github.com/simple-salesforce/simple-salesforce/pull/259 and https://github.com/simple-salesforce/simple-salesforce/pull/281)

@itymch would love to see this get rebased as this may be the closest one to the finish line?

**Comment by jon-wobken** [comment]
Fixed in #259

**Pull request closed without merging**